src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt

```diff
@@ -0,0 +1,159 @@
+package io.snyk.plugin
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.components.service
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.events.*
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import io.snyk.plugin.services.SnykTaskQueueService
+import io.snyk.plugin.snykcode.core.AnalysisData
+import io.snyk.plugin.snykcode.core.RunUtils
+import io.snyk.plugin.snykcode.core.SnykCodeUtils
+import io.snyk.plugin.ui.toolwindow.SnykToolWindowPanel
+import java.util.function.Predicate
+
+class SnykBulkFileListener() : BulkFileListener {
+
+    // proceed added and changed files
+    override fun after(events: MutableList<out VFileEvent>) {
+//        cleanCaches(
+//            events, listOf(
+//                VFileContentChangeEvent::class.java,
+//                VFileMoveEvent::class.java,
+//                VFileCopyEvent::class.java,
+//                VFileCreateEvent::class.java
+//            )
+//        )
```

Comment:
Let's don't merge commented code

Reply:
we probably will need it in the future and if not commit commented code it will be lost. 
would be happy if there is better way to not lost this code.

Reply:
git stash or local branch with changes?